app/uk/gov/hmrc/securemessage/repository/SecureMessageRepository.scala

```diff
@@ -76,11 +76,14 @@ abstract class SecureMessageRepository[A: TypeTag, ID](
   protected def getMessagesCount(identifiers: Set[Identifier], tags: Option[List[FilterTag]])(
     implicit ec: ExecutionContext): Future[Count] = {
     val querySelector = messagesQuerySelector(identifiers, tags)
-
     val totalCount: Future[Long] = getCount(querySelector)
     val unreadCount: Future[Long] = totalCount.flatMap { total =>
-      if (total == 0) { Future.successful(0.toLong) } else {
-        getCount(Json.obj("$and" -> Json.arr(querySelector, countUnreadQuery)))
+      if (total == 0) {
+        Future.successful(0.toLong)
+      } else if (collectionName == "conversation") {
```

Comment:
Feels like a weird and confusing if statement. What if the collectionName is conversation and the total is 0?